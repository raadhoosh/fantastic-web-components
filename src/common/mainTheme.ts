import { Theme } from './styled';
import * as Color from 'color';
function darken(hex: string) {
    return Color(hex)
        .darken(0.5)
        .hex();
}
const mainTheme: Theme = {
    color: {
        primary: '#00aba9',
        primary_dark: darken('#00aba9'),
        secondary: '#0066b1',
        secondary_dark: darken('#0066b1'),
        warning: '#f0ad4e',
        warning_dark: darken('#f0ad4e'),
        light: '#f8f9fa',
        light_dark: darken('#f8f9fa'),
        dark: '#000',
        success: '#d4edda',
        success_dark: '#155724',
        info: '#5bc0de',
        info_dark: darken('#5bc0de'),
        danger: '#d9534f',
        danger_dark: darken('#d9534f'),
    },
    font: {
        size: 16,
        family: '"Roboto"',
    },
    bottom: {
        fontsize: '1rem',
        border: '5px',
    },
    darken: (_color?: string) => {
        return Color(_color)
            .darken(0.03)
            .hex();
    },
    hexToRgba: (hex: string, a: number) => {
        return Color(hex).fade(a ? a : 0.5).string();
    },
};

export default mainTheme;
