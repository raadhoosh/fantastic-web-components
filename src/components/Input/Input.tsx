import styled from '../../common/styled';
import * as React from 'react';

interface IProps {
    children?: any;
    backgroundColor?: string;
    color?: string;
    margin?: string;
    small?: boolean;
    large?: boolean;
    noShadow?: boolean;
    theme?: any;
}

const EmInput = styled('input')(
    (props: (IProps)) => {
        const css = {
            'cursor': 'auto',
            'display': 'block',
            // 'width': '100%',
            'padding': '.375rem .75rem',
            'font-size': '1rem',
            'line-height': '1.5',
            'color': '#495057',
            'background-color': '#fff',
            'background-clip': 'padding-box',
            'border': `1px solid ${props.theme.color.primary}`,
            'border-radius': '.25rem',
            'transition': 'border-color .15s ease-in-out,box-shadow .15s ease-in-out',
            '&:focus': {
                'color': '#495057',
                'background-color': '#fff',
                'border-color': props.theme.color.primary,
                'outline': 0,
                'box-shadow': !props.noShadow ? `0 0 0 0.2rem ${props.theme.hexToRgba(props.theme.color.primary, 0.25)}` : 'none',
            }
        };
        if (props.small) {
            css.padding = '.25rem .5rem';
            css['font-size'] = '.875rem';
            css['line-height'] = '1.5';
            css['border-radius'] = '.2rem';
        }
        if (props.large) {
            css.padding = '.5rem 1rem';
            css['font-size'] = '1.25rem';
            css['line-height'] = '1.5';
            css['border-radius'] = '.3rem';
        }
        return css;
    }
);
/**
 * General component description in JSDoc format. Markdown is *supported*.
 */
const Input = (props: IProps) => {

    return (
        <EmInput {...props}/>
    );
};

export default Input;
