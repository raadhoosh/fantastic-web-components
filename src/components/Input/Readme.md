Input example:

```js
<Input small placeholder="First Name" />
<br/>
<Input placeholder="First Name" />
<br/>

<Input large placeholder="First Name" />

```