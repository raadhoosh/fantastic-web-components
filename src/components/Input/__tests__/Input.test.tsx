import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

// Add the custom matchers provided by 'jest-emotion'
expect.extend(createMatchers(emotion));

import Input from '../Input';
import mainTheme from '../../../common/mainTheme';

test('renders with correct styles', () => {
    const inp = renderer.create(<Input theme={mainTheme} value="text" small />).toJSON();
    if (inp) {
        expect(inp.props.value).toEqual('text');
    }
    expect(inp).toMatchSnapshot();
});
