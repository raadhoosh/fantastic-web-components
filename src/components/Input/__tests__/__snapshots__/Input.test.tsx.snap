// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
<input
  className="css-1gb2z4x"
/>
`;
